content/frontend/react-apollo/7-filtering-searching-the-list-of-links.md

```diff
@@ -159,74 +189,121 @@ const FEED_SEARCH_QUERY = gql`
+module.exports = {
+  feed
+};
+```
 
-That's the purpose of the [`withApollo`](http://dev.apollodata.com/react/higher-order-components.html#withApollo) function. This function injects the `ApolloClient` instance that you created in `index.js` into the `Search` component as a new prop called `client`.
+> **Note**: To understand what's going on in this resolver,
+> check out the
+> [Node tutorial](https://www.howtographql.com/graphql-js/0-introduction/).
 
-This `client` has a method called `query` which you can use to send a query manually instead of using the `graphql` higher-order component.
+In this case, two `where` conditions are specified: A link
+is only returned if either its `url` contains the provided
+`filter` _or_ its `description` contains the provided
+`filter`. Both conditions are combined using Prisma's `OR`
+operator.
 
-<Instruction>
+Perfect, the query is defined! But this time we actually
+want to load the data every time the user hits the **OK**
+button, not upon the initial load of the component. To do
+this, we'll use a hook supplied by Apollo called
+`useLazyQuery`. This hook performs a query in the same way
+the `useQuery` hook does but the difference is that it must
+be executed manually. This is perfect for our situation––we
+want to execute the query when the **OK** button is clicked.
 
-Here's what the code looks like. Open `Search.js` and implement `_executeSearch` as follows:
+Let's include `useLazyQuery` and execute it when the **OK**
+button is clicked.
 
 ```js(path=".../hackernews-react-apollo/src/components/Search.js")
-_executeSearch = async () => {
-  const { filter } = this.state
-  const result = await this.props.client.query({
-    query: FEED_SEARCH_QUERY,
-    variables: { filter },
-  })
-  const links = result.data.feed.links
-  this.setState({ links })
-}
+const Search = () => {
+  const [searchFilter, setSearchFilter] = useState('');
+  const [executeSearch, { data }] = useLazyQuery(
+    FEED_SEARCH_QUERY
+  );
+  return (
+    <>
+      <div>
+        Search
+        <input
+          type="text"
+          onChange={(e) => setSearchFilter(e.target.value)}
+        />
+        <button
+          onClick={() =>
+            executeSearch({
+              variables: { filter: searchFilter }
+            })
+          }
+        >
+          OK
+        </button>
+      </div>
+      {data &&
+        data.feed.links.map((link, index) => (
+          <Link key={link.id} link={link} index={index} />
+        ))}
+    </>
+  );
+};
 ```
 
-</Instruction>
+We'll want to be sure to import `useLazyQuery` at the top of
+the file.
 
-The implementation is almost trivial. You're executing the `FEED_SEARCH_QUERY` manually and retrieving the `links` from the response that's returned by the server. Then these links are put into the component's `state` so that they can be rendered.
+```js(path=".../hackernews-react-apollo/src/components/Search.js")
```

Comment:
This was imported in the first Instruction block in this tutorial

Reply:
can you clarify? I only see it being imported later down after it has been defined.

Reply:
`useLazyQuery` was imported in the beginning of the chapter in the first Instruction block. Here's a screenshot of the section.
![image](https://user-images.githubusercontent.com/33921841/100996505-2c3e0680-356a-11eb-99b9-36f0b619f9f0.png)